designforsf/brigade-matchmaker
# Add vagrant functionality

To test:
1) Install vagrant
https://www.vagrantup.com/docs/installation/
2) You probably also need to install virtualbox if you don't already have it
3) Open up a console and go to the source directory
4) `vagrant up`

Wait for the process to complete. When it says something like "Provisioning complete", you should be able to access the web service at localhost:5465.

Modifications to the source should be mirrored in the VM. However you may have to restart the node service.

Note that vagrant ssh moves the terminal into the guest VM. You can get out with "exit" once.

```
vagrant ssh
sudo systemctl restart brigade-matchmaker.target
```

If you change packages, the easiest way to trigger an npm install is to reprovision the VM:
```
vagrant provision
```
However this is slower than need be (even though it runs npm install for all components in parallel).

Alternatively, npm installing for a specific component inside the VM is a little clunky:
```
vagrant ssh
NODE_VERSION=6.12.2 /opt/nvm/nvm-exec npm --prefix /opt/brigade-matchmaker/components/web install
```
This is an area where the VM could be improved.

You can probably get away with npm install in the directory of the component on the host (ie outside the VM) as long as you are using the same npm version as the guest OS (currently 6.12.2, the latest on that LTS).
